app/models/playlist_item.rb

```diff
@@ -28,17 +28,19 @@ class PlaylistItem < ActiveRecord::Base
   end
 
   def duplicate!
-    new_clip = clip.dup
-    new_clip.save!
+    unless (clip.master_file.nil?)
```

Comment:
Instead of wrapping this all in an unless, how about:
`return nil if clip.master_file.nil?`